pkg/koordlet/statesinformer/states_noderesourcetopology.go

```diff
@@ -327,24 +327,41 @@ func (s *nodeTopoInformer) reportNodeTopology() {
 		for k, v := range nodeTopoAnnotations {
 			nodeResourceTopology.Annotations[k] = v
 		}
-		// TODO only update if necessary
-		s.updateNodeTopo(nodeResourceTopology)
 
-		// do UPDATE
-		if features.DefaultKoordletFeatureGate.Enabled(features.NodeTopologyReport) {
-			_, err = s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Update(context.TODO(), nodeResourceTopology, metav1.UpdateOptions{})
-			if err != nil {
-				klog.Errorf("failed to update cpu info of node %s, err: %v", node.Name, err)
-				return err
+		if s.isSyncNeeded(nodeResourceTopology.Annotations) {
+			// TODO only update if necessary
+			s.updateNodeTopo(nodeResourceTopology)
+
+			// do UPDATE
+			if features.DefaultKoordletFeatureGate.Enabled(features.NodeTopologyReport) {
+				_, err = s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Update(context.TODO(), nodeResourceTopology, metav1.UpdateOptions{})
+				if err != nil {
+					klog.Errorf("failed to update cpu info of node %s, err: %v", node.Name, err)
+					return err
+				}
 			}
+			return nil
 		}
 		return nil
 	})
 	if err != nil {
 		klog.Errorf("failed to update NodeResourceTopology, err: %v", err)
 	}
 }
-
+func (s *nodeTopoInformer) isSyncNeeded(newAnnotations map[string]string) bool {
+	nodename := s.nodeInformer.GetNode().Name
+	if s.nodeTopology == nil {
+		return true
+	}
+	if util.IsequalTopo(s.nodeTopology.Annotations, newAnnotations) {
```

Comment:
Except that koordlet needs to check whether it is equal before reporting, is there any place where this function needs to be reused? If not, consider putting it near the reporting code first.

Reply:
In addition, should we also consider supporting Spec?